### PR TITLE
Final changes to DSWx-HLS SAS for v1.0

### PR DESCRIPTION
**Pull request by gshiroma**
This PR provides final changes to the DSWx-HLS SAS for v1.0:
- Set default for `apply_ocean_masking` (in the default runconfig file) to `False`;
- Updates DSWx-HLS metadata:
    - Adds `AEROSOL_CLASS_REMAPPING_ENABLED` and `OCEAN_MASKING_ENABLED`;
    - Substitutes fields with value `-` by more meaningful values (addresses @nemo794 comment [here](https://github.com/nasa/PROTEUS/pull/38#discussion_r1127180356);
- Update URL pointing to the updated golden dataset in Zenodo (to be updated after all changes are approved).

### CODE REVIEW

File: src/proteus/dswx_hls.py [hunk 1]
```diff
@@ -3943,14 +3953,17 @@ def _populate_dswx_metadata_processing_parameters(
         dswx_metadata_dict['FOREST_MASK_LANDCOVER_CLASSES'] = \
             ','.join([str(c) for c in forest_mask_landcover_classes])
     else:
-        dswx_metadata_dict['FOREST_MASK_LANDCOVER_CLASSES'] = '-'
+        dswx_metadata_dict['FOREST_MASK_LANDCOVER_CLASSES'] = 'EMPTY'
+
+    # ocean masking
+    dswx_metadata_dict['OCEAN_MASKING_ENABLED'] = \
+        'TRUE' if apply_ocean_masking else 'FALSE'
```
Comment (nemo794) [inline src/proteus/dswx_hls.py:3960]: Please check my understanding: If a shoreline shapefile is not provided, then ocean masking will not be applied, regardless of how `apply_ocean_masking` is set.

If so, then in that scenario, should `OCEAN_MASKING_ENABLED` also be set to `False`? Currently, it looks like the code will set it to `TRUE`?

File: src/proteus/defaults/dswx_hls.yaml [hunk 2]
```diff
@@ -67,7 +67,7 @@ runconfig:
             check_ancillary_inputs_coverage: True
 
             # Apply ocean masking
-            apply_ocean_masking: True
+            apply_ocean_masking: False
```
Comment (nemo794) [inline src/proteus/defaults/dswx_hls.yaml:70]: Nice!

File: src/proteus/dswx_hls.py (hunk 1 above)
Comment (gshiroma) [reply src/proteus/dswx_hls.py:3960]: This is verified in `_check_ancillary_inputs()` if `apply_ocean_masking` is set but a shoreline shapefile is not provided, the SAS will raise an error in that function (btw, I'm can simplify the next condition `if apply_ocean_masking and shoreline_shapefile:`).

**Comment by hfattahi** [comment]
The changes look good to me. I will wait for a real data test

**Review by nemo794 — APPROVED** [review]
Using all defaults, the code appears to work and produce the desired outputs on multiple test datasets. The additional metadata fields are included, and the `NOT_PROVIDED` entries populate correctly. Nice!

Looking ahead, it would be beneficial to discuss and make sure everyone is in agreement on the behavior for when a subset of the three primary ancillary files (Worldcover, Landcover, and DEM) are missing. (The shoreline shapefile is handled differently.) However, for Ops processing, those three ancillary files will be included, and in that nominal case the code works well.

**Comment by gshiroma** [comment]
> Using all defaults, the code appears to work and produce the desired outputs on multiple test datasets. The additional metadata fields are included, and the `NOT_PROVIDED` entries populate correctly. Nice!
> 
> Looking ahead, it would be beneficial to discuss and make sure everyone is in agreement on the behavior for when a subset of the three primary ancillary files (Worldcover, Landcover, and DEM) are missing. (The shoreline shapefile is handled differently.) However, for Ops processing, those three ancillary files will be included, and in that nominal case the code works well.

Thank you, @nemo794 !

**Comment by gshiroma** [comment]
@hfattahi , just letting you know that I finished the changes to this PR

**Review by hfattahi — APPROVED** [review]
Thank you @gshiroma !

**Pull request merged**